lib/linters/analytics_event_name_linter.rb

```diff
@@ -4,16 +4,298 @@ module IdentityIdp
+          e4173d8b436e2137929fdb089a01be9c
+          fa5332bf24780a0e5d97c92cb6685c20
+          f94ad46e34cb1ef634dc56d0e07fac42
+          96db2d8c30e9619bb2b7f6c405d49d23
+          95d1a06b2eb9d67cbe8a5748ca46b7ae
+          1aa4b09d378ba47b8c5cfb601e669bd2
+          ccbc838eb8c5cdfc47bf51b06fc8a467
+          9714e092ac2a33bc9bab0c62e9ace573
+          3fc806f7fdf20755bc6fac01bd0d8c86
+          fc9ea0c4c32531bc098132f634994c71
+          2f08a61ab58af71a498333fa3b4604a0
+          d26d285287cf63ffc2f2e9cd8b939270
+          ce78de1891b10c24bb6d08910d81a3d9
+          1a814b6ad445ef51298d625a4f4950c7
+          8d4127f5138748415d9dc678068af6a8
+          a38d7132b9defea6c89ace62391eeb57
+          6f9ca12902af4b14d25b93e74a57988d
+          faa5adfbf1022ce65fa909576e93df43
+          e1aac4baf7a7dfd4b633453e0f5fc364
+          cd4c1afd502e3e603895adc867c464fc
+          de7251b793eff00fb207a0eee24177c5
+          6493d9645968d25443a7bf01ccfb2136
+          7a30fac0912f259310128a26b84bfab1
+          3bcec60b58b88c4515e10f76af85f4fa
+          93dfdf0013024ae5877bdf30883ed436
+          fd6bd3c66503a0a142bcdd34f9c22f91
+          8335775d256771c01989cbf1442d2cf8
+          ac1176cd29cf792e6965ee936f05c18a
+          6a79836cf56255c9865e45086206fae2
+          784f40a9ddbe25409cd68f9fcacac82c
+          307f8eff4a32d2b823faad2decd0d2ee
+          f21600d75621253ed14e7e4f5606d4d9
+          189c1bf5286214c54ddfcad6d256d1de
+          cace6811571d9c29b77e089c3c3d44f2
+          17e0c61cd3044e1da61480a180702236
+          a71de38cacaa3a3240cc7289da874213
+          43d6156f57c1a3bf858a905219bae445
+          66a2b2d42bb6ed9d018564f06b8e8f28
+          88c7243619e7d80e5ccf100904132435
+          93818ddf5a70310818e741ce5ce067f6
+          393c30caaaf0c4f0314ba3c38c13f63e
+          eea17f7001343fb794b04d3fb8a33be5
+          7e17e9d5f27c558a4f414daeceb3660b
+          3631bdb68856825b51e0b4c3273766f4
+          430f6b10170a3f9f1bc5a7f84dd8e948
+          002849f3f32a2adfb7829afba84f2400
+          8f19a1500f3a002be79737c9f66aae73
+          afe1a0413151cb4c52f76bc41930f3ef
+          5eaaedb4907b1320ea4ad42fe35a25a6
+          79b8cf1c4fc0a9b29561f5d7e7322094
+          7126d6be494165dafa0362c65067ecce
+          862a03bc38ccd59ff156b4cbdc974075
+          8bea4d5f3884117e71fc9f0183163717
+        ].to_set.freeze
+
         def on_send(node)
           first_argument, = node.arguments
           expected_name = ancestor_method_name(node)
-          if first_argument.type != :sym || first_argument.value != expected_name
-            add_offense(
-              first_argument,
-              location: :expression,
-              message: "Event name must match the method name, expected `:#{expected_name}`",
-            )
-          end
+          return if first_argument.type == :sym && first_argument.value == expected_name
+          return if LEGACY_EVENT_NAMES.include?(Digest::MD5.hexdigest(expected_name.to_s))
```

Comment:
This is a little bit of intentional obfuscation to hopefully add a little extra friction to someone's inclination to include their new event in the list above.

Reply:
I was trying to think of alternatives to this, some ideas:
- put the list in a `.yml` file with a similar comment
- maybe digest the whole YML file and check in the one digest that it hasn't been changed?

anyways long way to say, it's fine as is, it's going to be annoying anyways and is annoying on purpose.

Reply:
Hmm, I was thinking through some of those ideas. I like one digest, but I'm not sure how it'd work to consider whether an otherwise-flagged lint finding is "included" in that digest? The only thing I could imagine is aggregating all failures and then creating a digest from the set of failures, and expecting that the digest matches a singular value. The problem with that is if someone were to add a new failure, I don't know how the logic would be able to determine which is the "new' failure to give the developer a helpful error message.

Reply:
I went with a slightly different approach in b2a793c4743b903be00b7b82373f3578130f8444. Same idea, just checking only a portion of the hash, and collapsing the truncated digests to make it shorter. A little higher risk of a collision, but I wouldn't think it should be a big deal.

Reply:
If we're at all worried about collisions, let's not do MD5 then?

Reply:
I'm not really worried about collisions, and if a collision does happen, it's not a big deal. I chose MD5 for speed.